### PR TITLE
[Contract] Implement set_price Function

### LINKED ISSUE #508: [Contract] Implement `set_price` Function
## 📝 Description

Implement the `set_price` function that allows an authorized price feeder (admin) to push a new price for a given asset into the oracle's persistent storage.

**Depends on:** Initialize Oracle Adapter Contract Storage

## ✅ Requirements

- Signature: `fn set_price(env: Env, asset: Address, price: i128, timestamp: u64)`
- Require auth from the stored admin address
- Assert `price > 0` — reject zero or negative prices
- Assert `timestamp > 0`
- Construct `PriceData { price, timestamp }` and write to `Price(asset)` storage key
- Emit `PriceUpdated { asset, price, timestamp }` contract event

## 🎯 Acceptance Criteria

- [ ] Only admin can push prices
- [ ] Zero or negative price panics with clear error
- [ ] `PriceData` correctly stored and retrievable via `get_price`
- [ ] `PriceUpdated` event emitted with correct fields

## 📁 Expected files to change

- `contracts/oracle-adapter/src/lib.rs`
- `contracts/oracle-adapter/src/storage.rs`

---
> 🌿 Branch: `feat/oracle-set-price` | Priority: HIGH | Difficulty: ☕ one-coffee

---
**Priority:** HIGH · **Difficulty:** easy · **Track:** Contract

### PR DESCRIPTION
**Pull request by shoaib050326**
## 📝 Changes description

Implements the `set_price` function on the Oracle Adapter contract as specified in issue #508.

### Changes made:

1. **`contracts/oracle-adapter/src/storage.rs` (new)** — DataKey enum (`Admin`, `Price(Address)`) and helper functions for admin and price storage.

2. **`contracts/oracle-adapter/src/events.rs` (new)** — `PriceUpdated` contract event with `asset`, `price`, and `timestamp` fields.

3. **`contracts/oracle-adapter/src/lib.rs`** — 
   - Added `initialize(admin)` to set the contract admin (one-time, with auth).
   - Updated `get_price` to read from the `Price(asset)` DataKey.
   - Updated `set_price` to:
     - Require auth from the stored admin address.
     - Assert `price > 0` (rejects zero/negative).
     - Assert `timestamp > 0`.
     - Write `PriceData` to `Price(asset)` storage.
     - Emit `PriceUpdated` event.

4. **`contracts/oracle-adapter/src/test.rs`** — 10 comprehensive tests covering initialization, admin auth, price/timestamp validation, event emission, multi-asset storage, and edge cases.

5. **`contracts/collateral-vault/src/lib.rs`** — Fixed pre-existing build error (accessing field on `Option<Position>` instead of unwrapping) that was blocking CI on the workspace-level check.

### ✅ Acceptance Criteria

- [x] Only admin can push prices — verified via `require_auth` on stored admin
- [x] Zero or negative price panics with clear error — `assert!(price > 0, "price must be positive")`
- [x] `PriceData` correctly stored and retrievable via `get_price`
- [x] `PriceUpdated` event emitted with correct fields

## ⏰ Time spent breakdown

~30 minutes

Closes #508

### CODE REVIEW

**Review by ryzen-xp — CHANGES_REQUESTED** [review]
hello  please  remove  test snapshot and pass ci

**Review by ryzen-xp — CHANGES_REQUESTED** [review]
remove test snapshot and fix you pr code  that failing  ci

**Comment by ryzen-xp** [comment]
hello @shoaib050326  , can you do in  2 hour if  you can't  then i will unassign

**Comment by shoaib050326** [comment]
Hi @ryzen-xp working on it.

File: contracts/oracle-adapter/src/lib.rs [hunk 1]
```diff
@@ -49,8 +47,19 @@ impl OracleContract {
     pub fn set_price(env: Env, asset: Address, price: i128, timestamp: u64) {
         let caller = storage::get_admin(&env).expect("NotInitialized");
         caller.require_auth();
+
+        assert!(price > 0, "price must be positive");
```
Comment (ryzen-xp) [inline contracts/oracle-adapter/src/lib.rs:54]: use  contracterror type not hardcoded

**Review by ryzen-xp — CHANGES_REQUESTED** [review]
please remove test snapshots

File: contracts/oracle-adapter/src/lib.rs (hunk 1 above)
Comment (shoaib050326) [reply contracts/oracle-adapter/src/lib.rs:54]: fixed

**Review by ryzen-xp — APPROVED** [review]
